Bottom: 91153ae24e0243a1f3cf380e415d9bc5e8094789
Top:    c076dedd7fc3859a921947137ceb8b57fa008eb2
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-07-09 14:58:14 -0700

rsocket: Build rspreload library as part of build

Build rspreload.so, but do not install it.  This avoids the
need for the user to build it manually, without risking that
the preload library intercepts socket calls without the user's
explicit knowledge.

Signed-off-by: Sean Hefty <sean.hefty@intel.com>


---

diff --git a/Makefile.am b/Makefile.am
index 51b2f89..b734d45 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,11 +1,12 @@
 INCLUDES = -I$(srcdir)/include
 
-lib_LTLIBRARIES = src/librdmacm.la
+lib_LTLIBRARIES = src/librdmacm.la src/rspreload.la
 
 ACLOCAL_AMFLAGS = -I config
 AM_CFLAGS = -g -Wall -D_GNU_SOURCE -DSYSCONFDIR=\"$(sysconfdir)\" -DRDMADIR=\"@rdmadir@\"
 
 src_librdmacm_la_CFLAGS = $(AM_CFLAGS)
+src_rspreload_la_CFLAGS = $(AM_CFLAGS) -shared -fPIC
 
 if HAVE_LD_VERSION_SCRIPT
     librdmacm_version_script = -Wl,--version-script=$(srcdir)/src/librdmacm.map
@@ -19,6 +20,8 @@ src_librdmacm_la_LDFLAGS = -version-info 1 -export-dynamic \
 			   $(librdmacm_version_script)
 src_librdmacm_la_DEPENDENCIES =  $(srcdir)/src/librdmacm.map
 
+src_rspreload_la_SOURCES = src/preload.c src/indexer.c
+
 bin_PROGRAMS = examples/ucmatose examples/rping examples/udaddy examples/mckey \
 	       examples/rdma_client examples/rdma_server examples/rdma_xclient \
 	       examples/rdma_xserver examples/rstream examples/rcopy
